modules/workflow-workflowoperation/src/test/java/org/opencastproject/workflow/handler/workflow/DuplicateEventWorkflowOperationHandlerTest.java

```diff
@@ -0,0 +1,323 @@
+    Assert.assertEquals(Action.CONTINUE, result.getAction());
+
+    Track track = clonedMediaPackages.getValue().getTracksByTag("tag1")[0];
+    Assert.assertEquals("tag1", track.getTags()[0]);
+    Assert.assertEquals("tag2", track.getTags()[1]);
+  }
+
+  @Test
+  public void testRemoveAndAddTags() throws Exception {
+    mockDependencies(1);
+    Map<String, String> configurations = new HashMap<String, String>();
+    configurations.put(SOURCE_FLAVORS_PROPERTY,  "*/*");
+    configurations.put(SOURCE_TAGS_PROPERTY, "part1");
+    configurations.put(TARGET_TAGS_PROPERTY, "-part1,+tag3");
+    configurations.put(NUMBER_PROPERTY, "" + 1);
+    configurations.put(MAX_NUMBER_PROPERTY, "" + 10);
+    configurations.put(PROPERTY_NAMESPACES_PROPERTY, "org.opencastproject.assetmanager.security,org.opencastproject.scheduler");
+    configurations.put(COPY_NUMBER_PREFIX_PROPERTY, "copy");
+
+    // run the operation handler
+    WorkflowOperationResult result = getWorkflowOperationResult(mp, configurations);
+    Assert.assertEquals(Action.CONTINUE, result.getAction());
+
+    Track track = clonedMediaPackages.getValue().getTracksByTag("tag3")[0];
+    final List<String> newTags = Arrays.asList(track.getTags());
+    final  List<String> originalTags = Arrays.asList(mp.getTracksByTag("part1")[0].getTags());
+    Assert.assertEquals(originalTags.size(), newTags.size());
+    Assert.assertTrue(newTags.contains("tag3"));
+    Assert.assertFalse(newTags.contains("part1"));
+  }
+
+  private WorkflowOperationResult getWorkflowOperationResult(MediaPackage mp, Map<String, String> configurations)
+          throws WorkflowOperationException {
+    // Add the mediapackage to a workflow instance
+    WorkflowInstanceImpl workflowInstance = new WorkflowInstanceImpl();
+    workflowInstance.setId(1);
+    workflowInstance.setState(WorkflowState.RUNNING);
+    workflowInstance.setMediaPackage(mp);
+    WorkflowOperationInstanceImpl operation = new WorkflowOperationInstanceImpl("op", OperationState.RUNNING);
+    operation.setTemplate("create-event");
+    operation.setState(OperationState.RUNNING);
+    for (String key : configurations.keySet()) {
+      operation.setConfiguration(key, configurations.get(key));
+    }
+
+    List<WorkflowOperationInstance> operationsList = new ArrayList<WorkflowOperationInstance>();
+    operationsList.add(operation);
+    workflowInstance.setOperations(operationsList);
+
+    // Run the media package through the operation handler
+    return operationHandler.start(workflowInstance, null);
+  }
+
+  private void mockDependencies(int numberOfCopies) throws Exception {
+    clonedMediaPackages = Capture.newInstance(CaptureType.ALL);
+    reset(ingestService, workspace, assetManager, distributionService);
+
+
+    Capture<MediaPackage> mpCapt = Capture.newInstance();
+    expect(ingestService.addCatalog(anyObject(), anyString(), eq(MediaPackageElements.EPISODE), capture(mpCapt))).andAnswer(
```

Comment:
@lkiesow is there a policiy regarding the usage of java8 features in opencast? I would love to use streams, lambdas, CompletableFutures and so on, but I thought that there is some kind of rejection in the community. Hence, I avoided to use Java8 features. Am I free to use them?

Reply:
@krinnewitz There is no such policy and Opencast 4.x and higher officially require Java 8. I don't think there is any reason why you would need to avoid Java 8 features... feel free to use them.

Reply:
@krinnewitz ... but there is a code style guide saying "no more than 120 characters per line", see https://docs.opencast.org/develop/developer/code-style/

Reply:
hmm, looks like our version of felix doesn't work with java8 features which are not byte code compatible with java7. During build, I get:

Error building bundle org.opencastproject:opencast-workflow-workflowoperation:bundle:5-SNAPSHOT : Invalid class file: org/opencastproject/workflow/handler/workflow/DuplicateEventWorkflowOperationHandler.class